src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/matchxpath/InputMatchXpathAvoidInstanceCreationWithoutVar.java

```diff
@@ -0,0 +1,15 @@
+//non-compiled with javac: Compilable with Java14
```

Comment:
Why Java14? I see only type inference here. It is available since Java10

Reply:
I took as an example some other test files, 

modified to Java10

Reply:
> Why Java14? I see only type inference here. It is available since Java10

@pbludov What is the policy on this? I am wondering because I faced the same decision in #9060. There is two different ways to approach this:
1. Use highest version number that code compiles on at the time of writing the test case.
2. Lowest possible version that code compiles on (basically targeting the release version that the specific feature that makes it non-compilable first appeared in). 

For #9060 , I went with #1, thinking that we are guaranteed that the code is compilable up to at least Java 15. Is it better to do#2?
I guess one day, when Checkstyle itself is compilable by newer versions of Java, we can move those test case files with versions <= new Checkstyle version to compilable?

Reply:
Reason of this comment is to define what CI job should try to compile it.
Exact minimum version was good approach when java was less frequent in releases and we just used next version. Right now situation is different. We will upgrade to jdk11, so we will skip 9 and 10. So on upgrade to use jdk11 as minimum required version for project compilation we will move all jdk9, jdk10, jdk11 to compilable folder.

So, it is always ok to use minimum version in this comment, as we have bunch of CIs to execute compilation on each version. We can use our next target jdk - 11.

As we have no limits in CI execution, I would keep minimum version required for Input.